python/ray/tests/test_basic.py

```diff
@@ -864,6 +864,27 @@ def method(self):
     assert ray.get([id1, id2, id3, id4]) == [0, 1, "test", 2]
 
 
+def test_submit_many(shutdown_only):
+    ray.init(num_cpus=2, num_gpus=2, resources={"Custom": 2})
+
+    @ray.remote
+    def f():
+        return 1
+
+    ray.get([f._remote(num_cpus=np.random.uniform()) for _ in range(100)])
+    ray.get([f._remote(num_gpus=np.random.uniform()) for _ in range(100)])
+    ray.get([
+        f._remote(resources={"Custom": np.random.uniform()})
+        for _ in range(100)
+    ])
+
+    assert ray.global_state.available_resources() == {
```

Comment:
This is not sufficient. `ray.global_state.available_resources()` is updated asynchronously, so you may need to run in a loop until this is true (or it times out, in which case you should raise an exception).